src/main/java/taxi/web/filter/AuthenticationFilter.java

```diff
@@ -0,0 +1,43 @@
+package taxi.web.filter;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Set;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpSession;
+
+public class AuthenticationFilter implements Filter {
+    private Set<String> allowedUrls = new HashSet<>();
+
+    @Override
+    public void init(FilterConfig filterConfig) {
+        allowedUrls.add("/login");
+        allowedUrls.add("/drivers/add");
+        allowedUrls.add("/registration");
+        allowedUrls.add("/help");
+        allowedUrls.add("/contacts");
+    }
+
+    @Override
+    public void doFilter(ServletRequest servletRequest, ServletResponse servletResponse,
+                         FilterChain filterChain)
+            throws IOException, ServletException {
+        HttpServletRequest hreq = (HttpServletRequest) servletRequest;
+        HttpServletResponse hresp = (HttpServletResponse) servletResponse;
+        HttpSession session = hreq.getSession();
+
+        Long driverId = (Long) session.getAttribute("driver_id");
+        if (driverId != null || (driverId == null && allowedUrls.contains(hreq.getServletPath()))) {
+            filterChain.doFilter(hreq, hresp);
+            return;
+        }
```

Comment:
```suggestion
        if (driverId != null ||  allowedUrls.contains(hreq.getServletPath())) {
            filterChain.doFilter(hreq, hresp);
            return;
        }
```

i does not metter driverId == null or not if allowedUrls.contains(hreq.getServletPath())